client/post-editor/editor-sidebar/header.jsx

```diff
@@ -22,19 +24,32 @@ import DraftsButton from 'post-editor/drafts-button';
 import PostCountsData from 'components/data/post-counts-data';
 import QueryPostTypes from 'components/data/query-post-types';
 
+function goBack() {
+	if ( typeof window !== 'undefined' ) {
+		window.history.back();
+	}
+}
+
 function EditorSidebarHeader( { typeSlug, type, siteId, showDrafts, toggleDrafts, allPostsUrl, toggleSidebar } ) {
 	const isCustomPostType = ( 'post' !== typeSlug && 'page' !== typeSlug );
 	const className = classnames( 'editor-sidebar__header', {
 		'is-drafts-visible': showDrafts,
 		'is-loading': isCustomPostType && ! type
 	} );
-
-	let allPostsLabel;
-	switch ( typeSlug ) {
-		case 'post': allPostsLabel = translate( 'Posts' ); break;
-		case 'page': allPostsLabel = translate( 'Pages' ); break;
-		default: allPostsLabel = get( type, 'labels.menu_name' ) || translate( 'Loading…' );
+	const useBackButton = page.len > 0;
```

Comment:
Correct, that's the new behavior. I'm not in love with it, but I don't hate it either. Ideally we'd be able to contextualize the label to what you're heading back to, but I couldn't figure out how to weave that into the information available to the sidebar header.